netcore/System.Private.CoreLib/src/System/Array.Mono.cs

```diff
@@ -48,24 +54,30 @@ private class RawData
 			}
 		}
 
-		public static void Clear (Array array, int index, int length)
+		public static unsafe void Clear(Array array, int index, int length)
 		{
 			if (array == null)
-				ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
-			if (length < 0)
-				ThrowHelper.ThrowIndexOutOfRangeException();
+				ThrowHelper.ThrowArgumentNullException (ExceptionArgument.array);
 
-			int low = array!.GetLowerBound (0);
-			if (index < low)
-				ThrowHelper.ThrowIndexOutOfRangeException();
+			int lowerBound = array.GetLowerBound (0);
+			int elementSize = array.GetElementSize ();
+			nuint numComponents = (nuint)Unsafe.As<RawData> (array).Count;
 
-			index = index - low;
+			ref byte p = ref array.GetRawSzArrayData ();
+			RuntimeStructs.MonoClass* elementClass = RuntimeHelpers.GetMonoVTable (array)->klass->element_class;
 
-			// re-ordered to avoid possible integer overflow
```

Comment:
We don't want to expose MonoClass to managed code, you can call RuntimeTypeHandle.HasReferences (array.GetType ()). HasReferences can be converted to an intrinsics if needed.

Reply:
@vargaz well, if you think it's not a good idea I will re-write it back to an intrinsic for HasReferences, I noticed a trend to expose runtime structs to C# in CoreCLR and decided to try it in Mono too.

Reply:
`RuntimeTypeHandle.HasReferences (array.GetType ())` won't work because we need to check Array's Element Type for them.

Reply:
HasReferences is supposed to work for arrays correctly.